pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/FunctionsImpl.java

```diff
@@ -754,17 +721,19 @@ public void updateFunctionOnWorkerLeader(final String tenant,
         else{
             clsLoader = FunctionConfigUtils.validate(functionConfig, componentPackageFile);
         }
-        return FunctionConfigUtils.convert(functionConfig, clsLoader);
-
-    }
-
-    private static boolean hasPackageTypePrefix(String destPkgUrl) {
-        return Arrays.stream(PackageType.values()).anyMatch(type -> destPkgUrl.startsWith(type.toString()));
+        return clsLoader;
     }
 
-    private File downloadPackageFile(String packageName) throws IOException, PulsarAdminException {
-        File file = Files.createTempFile("function", ".tmp").toFile();
-        worker().getBrokerAdmin().packages().download(packageName, file.toString());
-        return file;
+    private ImmutablePair<String, String> getTypeClassName(String functionPackageUrl) {
+        try {
+            Map<String, String> properties = worker().getBrokerAdmin().packages()
```

Comment:
I don't understand what we are doing here.  We are not using pulsar admin to upload packages.

Reply:
We introduce [package management](https://docs.google.com/document/d/1FDSNhc8YB1yUMFBUrx8p-1nH1s5Kftp04PkFKWec-2U/edit) to manage function jars. 
We used to get the FunctionTypes by loading the jar and get it from the class. If we want to use a jar from the package management, we want to get it directly from the metadata and avoid download the jar at the broker.